### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by DRagsdaleCoding**
-Installed remaining modules needed for authentication
-set up login, register, and get current user routes
-set up hashing, salting, and encryption
-set up session serialization and deserialization
-set up working login and register on the app
-created user models and controllers for authentication routes
-added email field to the Users table to also serve as the  username
@maki-q 
@Gobleizer

### CODE REVIEW

**Review by Gobleizer — COMMENTED** [review]
Thank you for getting the Authentication Feature working
I am approving pending Quinton's approval of the interface

**Review by maki-q — APPROVED** [review]

**Pull request merged**
